src/liquid/components/ld-notification/test/ld-notification.spec.ts

```diff
@@ -0,0 +1,772 @@
+      jest.advanceTimersByTime(DEFAULT_NOTIFICATION_TIMEOUT)
+      await page.waitForChanges()
+
+      notifications = page.root.querySelectorAll(
+        '.ld-notification__item:not(.ld-notification__item--dismissed)'
+      )
+      expect(notifications.length).toEqual(1)
+    })
+
+    it('does not dismiss a notification of type "info" if it is in queue behind another notification', async () => {
+      jest.useFakeTimers()
+
+      const page = await newSpecPage({
+        components: [LdNotification],
+        html: `<ld-notification></ld-notification>`,
+      })
+      page.win.dispatchEvent(
+        new CustomEvent('ldNotificationAdd', {
+          detail: {
+            content: 'I am an info message.',
+            type: 'info',
+          },
+        })
+      )
+      await page.waitForChanges()
+
+      page.win.dispatchEvent(
+        new CustomEvent('ldNotificationAdd', {
+          detail: {
+            content: 'Ooops.',
+            type: 'alert',
+          },
+        })
+      )
+      await page.waitForChanges()
+
+      let notifications = page.root.querySelectorAll(
+        '.ld-notification__item:not(.ld-notification__item--dismissed)'
+      )
+      expect(notifications.length).toEqual(2)
+
+      // Fast-forward until the timer has been executed.
+      jest.advanceTimersByTime(DEFAULT_NOTIFICATION_TIMEOUT)
+      await page.waitForChanges()
+
+      notifications = page.root.querySelectorAll(
+        '.ld-notification__item:not(.ld-notification__item--dismissed)'
+      )
+      expect(notifications.length).toEqual(2)
+    })
+  })
+
+  describe('dismissing notifications', () => {
+    it('dismisses current notification on ldNotificationDismiss event', async () => {
+      const page = await newSpecPage({
+        components: [LdNotification],
+        html: `<ld-notification></ld-notification>`,
+      })
+      page.win.dispatchEvent(
+        new CustomEvent('ldNotificationAdd', {
+          detail: {
+            content: 'info #1',
+            type: 'info',
+          },
+        })
+      )
+      await page.waitForChanges()
+
+      page.win.dispatchEvent(
+        new CustomEvent('ldNotificationAdd', {
+          detail: {
+            content: 'info #2',
+            type: 'info',
+          },
+        })
+      )
+      await page.waitForChanges()
+
+      let notifications = page.root.querySelectorAll(
+        '.ld-notification__item:not(.ld-notification__item--dismissed)'
+      )
+      expect(notifications.length).toEqual(2)
+
+      page.win.dispatchEvent(new CustomEvent('ldNotificationDismiss'))
+      await page.waitForChanges()
+
+      notifications = page.root.querySelectorAll(
+        '.ld-notification__item:not(.ld-notification__item--dismissed)'
+      )
+      expect(notifications.length).toEqual(1)
+      expect(notifications[0].textContent).toContain('info #1')
+    })
+
+    it('does not do anything on ldNotificationDismiss event if no notifications are queued', async () => {
```

Comment:
Not sure if this test is actually necessary. When would you expect this test to fail?

Reply:
It tests the if condition [here](https://github.com/emdgroup-liquid/liquid/pull/8/files#diff-d07fd92aa7330370f8b59a595ae1cae03f96750b4b899387cad69c0dc525082dR87). Notifications can be dismissed programmatically and the if condition makes sure no code is executed if there is no notification in the queue.

Reply:
If the condition was gone, the test would throw an exception.